src/interface/viewport.c

```diff
@@ -1098,6 +1098,105 @@ int sub_98197C(
 	return 0;
 }
 
+static void vehicle_visual_roto_drop(int x, int imageDirection, int y, int z, rct_vehicle *vehicle, int rct2VehiclePtrFormat)
+{
+	//RCT2_CALLPROC_X(0x006D5DA9, x, imageDirection, y, z, (int)vehicle, rct2VehiclePtrFormat, 0); return;
+	const rct_ride_entry_vehicle *vehicleEntry = (const rct_ride_entry_vehicle *)(rct2VehiclePtrFormat + 0x1A);
+	int image_id = (vehicle->var_C5 / 4) & 0x3;
+	image_id += vehicleEntry->base_image_id;
+	image_id += 4;
+	if (vehicle->restraints_position >= 64) {
+		image_id += 7;
+		image_id += (vehicle->restraints_position / 64);
+	}
+	int eax = 0x29 << 8 | 0;
+	int ebx = image_id | (vehicle->colours.body_colour << 19 | vehicle->colours.trim_colour << 24 | 0xA0000000);
+	int ebp = get_current_rotation();
+	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_X, uint16) = 0xFFF5;
+	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_Y, uint16) = 0xFFF5;
+	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_Z, uint16) = z + 1;
+	switch (get_current_rotation()) {
```

Comment:
These functions have already been implemented before the switch. See sub_ 98199c and such (there are a couple different ones)